libsnark/zk_proof_systems/plonk/tests/test_plonk.cpp

```diff
@@ -1080,13 +1088,157 @@ template<typename ppT, class transcript_hasher> void test_plonk_verifier()
 template<typename ppT> void test_plonk_gates_matrix_transpose()
 {
     using Field = libff::Fr<ppT>;
-    // load gates matrix from example circuit
+    // Load gates matrix from example circuit.
     plonk_example example;
     std::vector<std::vector<Field>> gates_matrix_transpose =
         plonk_gates_matrix_transpose(example.gates_matrix);
     ASSERT_EQ(gates_matrix_transpose, example.gates_matrix_transpose);
 }
 
+// generic test for all curves
+template<typename ppT> void test_plonk_constants_k1_k2()
+{
+    ppT::init_public_params();
+
+    using Field = libff::Fr<ppT>;
+
+    // Check that plonk_generate_constants_k1_k2 generates valid k1
+    // and k2.
+    Field k1, k2;
+    plonk_generate_constants_k1_k2(k1, k2);
+    ASSERT_TRUE(plonk_are_valid_constants_k1_k2(k1, k2));
+
+    // Store the valid choices for k1,k2.
+    Field k1_valid = k1;
+    Field k2_valid = k2;
+    // Example value for the number of gates (must be power of 2).
+    size_t num_gates = 8;
+    // Generate the n-th root of unity used as a generator of the
+    // multiplicative group H of size num_gates.
+    Field omega_base = libff::get_root_of_unity<Field>(num_gates);
+
+    // Check that plonk_are_valid_constants_k1_k2 correctly detects
+    // invalid combinations of k1 and k2.
+    for (size_t i = 1; i <= num_gates; ++i) {
+        size_t ipower = i;
```

Comment:
why the extra variable?  Why not just use `for (size_t ipower ...`?

Reply:
Indeed. Fixed.